application/controllers/UploaderController.php

```diff
@@ -347,7 +352,11 @@ public function run($actionID)
 
         $fn = $sFieldName;
         $qid = (int) Yii::app()->request->getParam('qid');
-        $qidattributes = QuestionAttribute::model()->getQuestionAttributes($qid);
+        $oQuestion = Question::model()->find("sid = :sid and qid = :qid and type = :type", [":sid" => $surveyid, ":qid" => $qid, ":type" => Question::QT_VERTICAL_FILE_UPLOAD]);
```

Comment:
Here we have the same question load as we have seen earlier. While this is correct, we could create a method for finding the question and calling getQuestionAttributes for slightly shortening the code and making it easier to maintain.

Reply:
> Here we have the same question load as we have seen earlier. While this is correct, we could create a method for finding the question and calling getQuestionAttributes for slightly shortening the code and making it easier to maintain.

We already get the question ? I don't see it.

Reply:
Sorry for the confusion. I've just meant that we have two places where we load the question:

https://github.com/LimeSurvey/LimeSurvey/pull/3782/files#diff-da3ae6cdc4a22c11c24a0ed743eb4bc1a190312eda624342eff57cc1e27883eeR151

and

https://github.com/LimeSurvey/LimeSurvey/pull/3782/files#diff-da3ae6cdc4a22c11c24a0ed743eb4bc1a190312eda624342eff57cc1e27883eeR355

we do not load it twice in the same run, but we do have both in the code and separating this feature into a method and then calling the method from both places will make the code more readable and easier to maintain. Not a huge trouble really, but it's a small thing we can improve upon.